packages/dds/map/src/directory.ts

```diff
@@ -1096,20 +1093,33 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
 	private readonly _subdirectories: Map<string, SubDirectory> = new Map();
 
 	/**
-	 * Keys that have been modified locally but not yet ack'd from the server.
+	 * Keys that have been modified locally but not yet ack'd from the server. This is for operations on keys like
+	 * set/delete operations on keys. The value of this map is list of pendingMessageIds at which that key
+	 * was modified. We don't store the type of ops, and behaviour of key ops are different from behaviour of sub
+	 * directory ops, so we have separate map from pendingSubDirectories.
 	 */
 	private readonly pendingKeys: Map<string, number[]> = new Map();
 
 	/**
-	 * Subdirectories that have been created/deleted locally but not yet ack'd from the server.
+	 * Subdirectories that have been created/deleted locally but not yet ack'd from the server. This is for operations
+	 * like create/delete sub directories. The value of this map is list of pendingMessageIds at which that sub
+	 * directory was created/deleted. We don't store the type of ops, and behaviour of key ops are different from
+	 * behaviour of sub directory ops, so we have separate map from pendingSubDirectories.
 	 */
 	private readonly pendingSubDirectories: Map<string, number[]> = new Map();
 
 	/**
-	 * Subdirectories that have been deleted locally but not yet ack'd from the server. This maintains the count
-	 * of delete op that are pending or yet to be acked from server.
+	 * Subdirectories that have been deleted locally but not yet ack'd from the server. This maintains the record
+	 * of delete op that are pending or yet to be acked from server. This is maintained just to track the locally
+	 * deleted sub directory as pendingSubDirectories does not track that specifically.
+	 */
+	private readonly pendingDeleteSubDirectoriesTracker: Set<string> = new Set();
```

Comment:
I don't follow how logic in this area avoids the need to count number of times the directory may have been deleted/recreated while pending. Ex: client A disconnects and while disconnected, creates `/dir` and deletes `/dir` 3 times, then reconnects. I think there are still bugs in this area: if I check out this branch and modify fuzz test config to lean heavily towards creation/deletion of the same directory (by for example limiting the directory name pool to have 2 options and only generating subdirectory create/delete ops), I see lots of failures with json that resembles the above kind of thing. See [this commit](https://github.com/Abe27342/FluidFramework/commit/ff212cab4fafae2061795dd09a4faf07f529f494).

Reply:
resolved.